saleor/webhook/base.py

```diff
@@ -0,0 +1,67 @@
+from collections.abc import Iterable
+from functools import partial
+from typing import Any, Callable, Optional, TypeVar
+
+from ..core.middleware import Requestor
+from ..core.utils.events import call_event
+from ..permission.enums import BasePermissionEnum
+from .models import Webhook
+
+
+class WebhookBase:
+    description: str
+    event_type: str
+    name: str
+    permission: BasePermissionEnum
+    subscription_type: str
+
+
+class SyncWebhookBase(WebhookBase):
+    # For now this class is only use to distinguish between sync and async webhooks.
+    # When migrating more webhooks to the new approach, this class could implement
+    # generic function to trigger sync webhooks.
+    pass
+
+
+class AsyncWebhookBase(WebhookBase):
+    @classmethod
+    def legacy_payload_func(cls) -> Optional[Callable]:
+        return None
```

Comment:
I think `None` is OK since this method is optional and only certain webhooks will implement it. Also code easily handles cases when this returns `None`.